pkg/scheduler/framework/plugins/podtopologyspread/filtering.go

```diff
@@ -223,31 +225,42 @@ func (pl *PodTopologySpread) calPreFilterState(pod *v1.Pod) (*preFilterState, er
 		TpPairToMatchNum:     make(map[topologyPair]*int32, sizeHeuristic(len(allNodes), constraints)),
 	}
 	requiredSchedulingTerm := nodeaffinity.GetRequiredNodeAffinity(pod)
+
+	var availableNodes []*framework.NodeInfo
 	for _, n := range allNodes {
 		node := n.Node()
 		if node == nil {
 			klog.ErrorS(nil, "Node not found")
 			continue
 		}
+
+		// Filter the node out if it carries a `node.kubernetes.io/unschedulable:NoSchedule` taint and pod doesn't tolerate it
+		if taints.TaintExists(node.Spec.Taints, unschedulableTaint) && !corev1.TolerationsTolerateTaint(pod.Spec.Tolerations, unschedulableTaint) {
```

Comment:
what's the performance impact of running the TolerationsTolerateTaint inside the loop for N nodes vs outside the loop a single time?

Reply:
what's the rationale behind limiting this to `unschedulable`? isn't that just a special-case of the general problem of topologySpreadConstraints breaking scheduling in the presence of untolerated taints?

Reply:
> what's the performance impact of running the TolerationsTolerateTaint inside the loop for N nodes vs outside the loop a single time?

sorry liggitt, I can't fully understand the meaning of outside the loop a single time, do you mean we should find a way to avoid loops.

The taint filter is O(i), i is somehow defined by the larger amount of node taints or pod tolerations, but I think they won't be too big. Just some of my opinions.

Reply:
@kerthcet we may need to hold the PR review for now, until we get to a consensus on whether o add this behavior or not, and how. See the original issue for discussion.